karpenter-operator/controllers/nodeclass/ec2_nodeclass_controller.go

```diff
@@ -102,6 +102,10 @@ func (r *EC2NodeClassReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
+	if hcp.Annotations[hyperv1.KarpenterCoreE2EOverrideAnnotation] == "true" {
```

Comment:
this seems to be completely skipping reconciliation, any reason to not skip SetupWithManager completely?